Client/SafeBrowsing/GSafeBrowsing.swift

```diff
@@ -0,0 +1,316 @@
+                                    }
+        )
+        
+        do {
+            let body = FindRequest(client: clientInfo, threatInfo: threatInfo)
+            let request = try encode(.post, endpoint: .fullHashes, body: body)
+            executeRequest(request, type: FindResponse.self) { response, error in
+                if let error = error {
+                    return completion(false, error)
+                }
+                
+                if let response = response {
+                    if !response.matches.isEmpty {
+                        return completion(false, nil)
+                    }
+                }
+                
+                completion(true, nil)
+            }
+        } catch {
+            completion(false, error)
+        }
+    }
+    
+    func fetch(_ completion: @escaping (Error?) -> Void) {
+        let clientInfo = ClientInfo(clientId: SafeBrowsingClient.clientId,
+                                    clientVersion: SafeBrowsingClient.version)
+        
+        let constraints = Constraints(maxUpdateEntries: UInt32(SafeBrowsingClient.maxBandwidth),
+                                      maxDatabaseEntries: UInt32(SafeBrowsingClient.maxDiskSpace),
+                                      region: "US",
+                                      supportedCompressions: [.raw],
+                                      language: nil,
+                                      deviceLocation: nil)
+        
+        let lists = [
+            ListUpdateRequest(threatType: .malware,
+                              platformType: .any,
+                              threatEntryType: .url,
+                              state: database.getState(.malware),
+                              constraints: constraints),
+            
+            ListUpdateRequest(threatType: .socialEngineering,
+                              platformType: .any,
+                              threatEntryType: .url,
+                              state: database.getState(.socialEngineering),
+                              constraints: constraints),
+            
+            ListUpdateRequest(threatType: .unwantedSoftware,
+                              platformType: .any,
+                              threatEntryType: .url,
+                              state: database.getState(.unwantedSoftware),
+                              constraints: constraints)
+        ]
+        
+        do {
+            let body = FetchRequest(client: clientInfo, listUpdateRequests: lists)
+            let request = try encode(.post, endpoint: .fetch, body: body)
+            executeRequest(request, type: FetchResponse.self) { [weak self] response, error in
+                guard let self = self else { return }
+                
+                if let error = error {
+                    return completion(error)
+                }
+                
+                if let response = response {
```

Comment:
The response has `minimumWaitDuration` associated with every response, we should use that to update the db

Reply:
Fixed.